### PR TITLE
network: function to fetch the genesis block

### PR DESCRIPTION
**Pull request by mzabaluev**
Add a blocking function to connect to the first of the trusted
peers and fetch the block specified by hash.

### CODE REVIEW

File: src/network/mod.rs [hunk 1]
```diff
@@ -196,9 +197,14 @@ fn handle_propagation(
...
     input.for_each(move |msg| {
+        debug!("to propagate: {:?}", &msg);
         let channels = channels.clone();
         let state = state.clone();
-        let nodes = state.topology.view().collect();
+        let nodes = state.topology.view().collect::<Vec<_>>();
+        debug!(
+            "will propagate to: {:?}",
+            nodes.iter().map(|node| node.id()).collect::<Vec<_>>()
+        );
```
Comment (mzabaluev) [inline src/network/mod.rs]: These debug lines got swept in with the other changes.

File: src/network/mod.rs [hunk 2]
```diff
@@ -307,3 +320,23 @@ pub fn bootstrap(config: &Configuration, blockchain: BlockchainR) {
...
+/// The calling thread is blocked until the block is retrieved.
+/// This function is called during blockchain initialization
+/// to retrieve the genesis block.
+pub fn fetch_block(config: &Configuration, hash: &HeaderHash) -> Result<Block, ()> {
+    if config.protocol != Protocol::Grpc {
+        unimplemented!()
+    }
+    match first_trusted_peer_address(config) {
+        None => {
+            error!("no gRPC peers specified, cannot retrieve block {}", hash);
```
Comment (NicolasDP) [inline src/network/mod.rs]: can we have an error report instead ?

File: src/network/mod.rs [hunk 3]
```diff
@@ -307,3 +320,23 @@ pub fn bootstrap(config: &Configuration, blockchain: BlockchainR) {
...
+
+/// Queries the trusted peers for a block identified with the hash.
+/// The calling thread is blocked until the block is retrieved.
+/// This function is called during blockchain initialization
+/// to retrieve the genesis block.
+pub fn fetch_block(config: &Configuration, hash: &HeaderHash) -> Result<Block, ()> {
+    if config.protocol != Protocol::Grpc {
+        unimplemented!()
+    }
+    match first_trusted_peer_address(config) {
```
Comment (NicolasDP) [inline src/network/mod.rs]: This will do for now. But we need to have concurrent fetch instead in the long run.

File: src/network/grpc/client.rs [hunk 4]
```diff
@@ -67,3 +71,38 @@ fn subscribe(
             Ok((node_id, prop_handles))
         })
 }
+
+// Fetches a block from a network peer in a one-off, blocking call.
+// This function is used during node bootstrap to fetch the genesis block.
+pub fn fetch_block(peer: Peer, hash: &HeaderHash) -> Result<Block, ()> {
```
Comment (NicolasDP) [inline src/network/grpc/client.rs]: I'd rather like this function to return a proper error code.

You can now create new error type fairly easily:
```rust
custom_error! {pub FetchBlockError
    // ...
}

**Pull request merged**